src/build/__init__.py

```diff
@@ -156,14 +158,25 @@ def __init__(self, srcdir, python_executable=sys.executable):
 
         self._build_system = build_system
         self._backend = self._build_system['build-backend']
-
+        self.script_dir = os.path.dirname(python_executable) if scripts_dir is None else scripts_dir
         self._hook = pep517.wrappers.Pep517HookCaller(
             self.srcdir,
             self._backend,
             backend_path=self._build_system.get('backend-path'),
             python_executable=python_executable,
+            runner=self._runner,
         )
 
+    def _runner(self, cmd, cwd=None, extra_environ=None):
+        # type: (Sequence[str], Optional[Union[bytes, Text]], Optional[Dict[str, str]]) -> None
+        # if script dir is specified must be inserted at the start of PATH (avoid duplicate path while doing so)
+        paths = OrderedDict()  # type:  Dict[str, None]
+        paths[str(self.script_dir)] = None
+        paths.update((i, None) for i in os.environ.get('PATH', '').split(os.pathsep))
```

Comment:
This will add `''`, the current working directory, to the path if the path is undefined or the empty string, is that intentional?

Reply:
Also, why does the `PATH` need to be deduplicated?